multiraft/heartbeat_test.go

```diff
@@ -103,13 +103,17 @@ func (hc heartbeatCount) String() string {
 // heartbeatCountMap will contain the count of heartbeat requests and responses
 // for each nodeID observed in the message stream.
 func countHeartbeats(ch <-chan *interceptMessage,
-	cond heartbeatCondition) heartbeatCountMap {
+	cond heartbeatCondition, t *testing.T) heartbeatCountMap {
 
 	cnt := make(heartbeatCountMap)
-	processEventsUntil(ch, nil, func(req *RaftMessageRequest) bool {
-		from := cnt[req.Message.From]
-		to := cnt[req.Message.To]
-		switch req.Message.Type {
+	processEventsUntil(ch, nil, func(req *proto.RaftMessageRequest) bool {
+		msg, err := req.UnmarshalMsg()
+		if err != nil {
+			t.Fatal(err)
```

Comment:
might be better to `t.Error` here for pragmatism (`Fatal` causes a panic, so most likely the test is just going to get stuck, and sometimes you lose valuable information about the test failure that way)